eclair-core/src/test/scala/fr/acinq/eclair/io/PeerConnectionSpec.scala

```diff
@@ -459,5 +473,23 @@ class PeerConnectionSpec extends TestKitBaseClass with FixtureAnyFunSuiteLike wi
     peer.send(peerConnection, message)
     transport.expectMsg(message)
   }
+
+  test("filter private IP addresses") { _ =>
+    val testCases = Seq(
+      NodeAddress.fromParts("127.0.0.1", 9735).get -> false,
+      NodeAddress.fromParts("0.0.0.0", 9735).get -> false,
+      NodeAddress.fromParts("192.168.0.1", 9735).get -> false,
+      NodeAddress.fromParts("140.82.121.3", 9735).get -> true,
+      NodeAddress.fromParts("0000:0000:0000:0000:0000:0000:0000:0001", 9735).get -> false,
+      NodeAddress.fromParts("b643:8bb1:c1f9:0556:487c:0acb:2ba3:3cc2", 9735).get -> true,
+      NodeAddress.fromParts("hsmithsxurybd7uh.onion", 9735).get -> false,
+      NodeAddress.fromParts("iq7zhmhck54vcax2vlrdcavq2m32wao7ekh6jyeglmnuuvv3js57r4id.onion", 9735).get -> false,
```

Comment:
Onion addresses are public though?

Reply:
Yes, but the function we're testing checks for public IP addresses, I renamed the field in https://github.com/ACINQ/eclair/pull/1973/commits/0bdb7e6654285f4d2507399d46d4e05db9d16da7